retry/regular.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package retry
+
+import (
+	"time"
+
+	"github.com/juju/utils/clock"
+)
+
+// Regular represents a strategy that repeats at regular intervals.
+type Regular struct {
+	Total time.Duration // total duration of attempt.
+	Delay time.Duration // interval between each try in the burst.
+	Min   int           // minimum number of retries; overrides Total
```

Comment:
I find it weird to have a *minimum* number of retries, and not a maximum.  I get it, in combination with the time limit, but still, I'd also expect there to be a way to say "just try 3 times"... I guess min works for that, if Total is zero.... but that's really counterintuitive... and doesn't give you the way to say "run for a max of Total time or max retries, whichever comes first"

Also, "overrides Total" is not exactly correct  Override, to me, means "totally ignore" ... but what actually happens is, you run for a min number of retries or for Total time, whichever comes later.